src/discussions/posts/data/api.js

```diff
@@ -152,6 +154,7 @@ export async function updateThread(threadId, {
     following,
     closed,
     pinned,
+    edit_reason_code: editReasonCode,
```

Comment:
`snakeCaseObject` will take care of converting this. 
```suggestion
    editReasonCode,
```